pkg/asset/manifests/operators.go

```diff
@@ -150,65 +162,52 @@ func (m *Manifests) generateBootKubeManifests(dependencies asset.Parents) []*ass
-		kubeSystemConfigmapRootCA,
-		kubeSystemSecretEtcdClient,
-		openshiftConfigSecretEtcdMetricsClient,
-		openshiftConfigConfigmapEtcdMetricsServingCA,
-		openshiftMachineConfigOperator,
-		etcdServiceKubeSystem,
-		hostEtcdServiceKubeSystem,
-	)
-	assetData := map[string][]byte{
-		"kube-cloud-config.yaml":                                  applyTemplateData(kubeCloudConfig.Files()[0].Data, templateData),
-		"machine-config-server-tls-secret.yaml":                   applyTemplateData(machineConfigServerTLSSecret.Files()[0].Data, templateData),
-		"pull.json":                                               applyTemplateData(pull.Files()[0].Data, templateData),
-		"cvo-overrides.yaml":                                      applyTemplateData(cVOOverrides.Files()[0].Data, templateData),
-		"host-etcd-service-endpoints.yaml":                        applyTemplateData(hostEtcdServiceEndpointsKubeSystem.Files()[0].Data, templateData),
-		"kube-system-configmap-etcd-serving-ca.yaml":              applyTemplateData(kubeSystemConfigmapEtcdServingCA.Files()[0].Data, templateData),
-		"kube-system-configmap-root-ca.yaml":                      applyTemplateData(kubeSystemConfigmapRootCA.Files()[0].Data, templateData),
-		"kube-system-secret-etcd-client.yaml":                     applyTemplateData(kubeSystemSecretEtcdClient.Files()[0].Data, templateData),
-		"openshift-config-secret-etcd-metrics-client.yaml":        applyTemplateData(openshiftConfigSecretEtcdMetricsClient.Files()[0].Data, templateData),
-		"openshift-config-configmap-etcd-metrics-serving-ca.yaml": applyTemplateData(openshiftConfigConfigmapEtcdMetricsServingCA.Files()[0].Data, templateData),
-
-		"04-openshift-machine-config-operator.yaml": []byte(openshiftMachineConfigOperator.Files()[0].Data),
-		"etcd-service.yaml":                         []byte(etcdServiceKubeSystem.Files()[0].Data),
-		"host-etcd-service.yaml":                    []byte(hostEtcdServiceKubeSystem.Files()[0].Data),
+	assetData := map[string][]byte{}
+	for _, a := range []asset.WritableAsset{
+		&bootkube.CVOOverrides{},
+		&bootkube.EtcdServiceKubeSystem{},
+		&bootkube.HostEtcdServiceEndpointsKubeSystem{},
+		&bootkube.HostEtcdServiceKubeSystem{},
+		&bootkube.KubeCloudConfig{},
+		&bootkube.KubeSystemConfigmapEtcdCA{},
+		&bootkube.KubeSystemConfigmapEtcdServingCA{},
+		&bootkube.KubeSystemConfigmapRootCA{},
+		&bootkube.KubeSystemSecretEtcdClient{},
+		&bootkube.KubeSystemSecretEtcdClientCADeprecated{},
+		&bootkube.KubeSystemSecretEtcdSigner{},
+		&bootkube.KubeSystemSecretEtcdSignerClient{},
+		&bootkube.MachineConfigServerTLSSecret{},
+		&bootkube.OpenshiftConfigConfigmapEtcdMetricsServingCA{},
+		&bootkube.OpenshiftConfigSecretEtcdMetricsClient{},
+		&bootkube.OpenshiftMachineConfigOperator{},
+		&bootkube.Pull{},
+	} {
+		dependencies.Get(a)
+		for _, f := range a.Files() {
+			assetData[strings.TrimSuffix(filepath.Base(f.Filename), ".template")] = applyTemplateData(f.Data, templateData)
```

Comment:
This is a welcome change over what was being done. We don't need the `assetData` map as a temporary home. We can add the file directly to a `*assetFile` slice here.

Reply:
donw with https://github.com/openshift/installer/compare/c52894b2c2064e5cb8a6d736ccdbe29279d18ceb..53c6fc3c1af4567bff062424a0a5aef6d587d93f